hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java

```diff
@@ -2201,7 +2201,18 @@ public boolean isRecordIndexEnabled() {
    * @return {@code true} if the specified metadata index is marked for dropping, {@code false} otherwise.
    */
   public boolean isDropMetadataIndex(String indexName) {
-    return StringUtils.nonEmpty(getMetadataConfig().getMetadataIndexToDrop()) && getMetadataConfig().getMetadataIndexToDrop().equals(indexName);
+    String subIndexNameToDrop = getMetadataConfig().getMetadataIndexToDrop();
```

Comment:
is this subIndexNameToDrop is only going to be used for SI and EI cases after this patch ?

Reply:
I think we can eliminate the whole method.